DIP upload integration, refs #7754

jraddaoui opened the pull request:
Adds arRestApiPlugin:
- Used to fetch LOD from Archivematica for DIP uploads to AtoM
- Only with browse taxonomies endpoint
- Not enabled by default but available in the plugin list
- With authoritation fix for Apache

Adds hierarchical DIP upload method:
- Used if a logical structMap exists in the METS file

Changes non-hieararchical DIP upload method:
- It now uses the fileGrp element to create the IO instead of looking at the objects folder
- The IO are now saved only once (in both process)
- Helper function moved to Qubit class

Changes metsData ES mapping:
- Adds Mediainfo tracks and other data that was added for the DRMC project

Adds QubitMetsParser:
- Moved most of the METS related code in the DIP upload process to it
- Used to get the data needed from the METS file for the IO search index
- Code clean-ups in the ES related code
- Allow METS with and without the namespace changes

jraddaoui committed:
Add arRestApiPlugin

Only with the browse taxonomies endpoint

jraddaoui committed:
Fix API authorization in Apache

jraddaoui committed:
Modify comment

jraddaoui committed:
Changes in browse taxonomy endpoint

- Remove id from results
- Allow culture as a param
- Return note and name in requested culture or the default culture
with culture fallback
- Return only not empty fields

jraddaoui committed:
Add hierarchical DIP upload

If a structMap type logical is included in the METS file
the information objects are created following the parent-child
div hierarchy and digital objects are added to those divs
with a fptr tag as child

A level of description is added to the info objects from the
type attribute inside the div. If one of those levels doesn't
exit in AtoM the upload process is stopped

Fixes #6954

jhsimpson committed:
enabling arRestApiPlugin

refs #6954

arRestApiPlugin must be enabled in order for api urls to work

jraddaoui committed:
Allow enable/disable arRestApiPlugin, refs #6863

jraddaoui committed:
Fix LOD mapping if no TYPE attr, fixes #7098

jraddaoui committed:
Allows 'div' and 'mets:div' in logical structMap

jraddaoui committed:
Create file ID to UUID mapping

jraddaoui committed:
Check main DMD section only in physical structMap

jraddaoui committed:
Change DIP upload process from logical structMap

- divs without TYPE are not added and their children added to its parent
- LOD from div with LABEL='objects' added to target info object
- Creates UUID mapping for files under fileGrp with USE='original'
- Only adds files wich UUID is found in the objects folder

jraddaoui committed:
Changes in METS data for ES, refs #7754

- Adds Mediainfo tracks and some other data from DRMC
- Moves METS parser code to QubitMetsParser class
- Code clean-ups and cosmetic changes

jraddaoui committed:
Fix Mediainfo queries, refs #7754

- Workaround to allow old and new METS files from Archivematica
  after the namespace changes

jraddaoui committed:
Improve DIP upload code, refs #7754

- Move most of the code related to METS to QubitMetsParser
- Move helper function to Qubit class
- Change non-hierarchical method to use fileGrp (USE="Original")
  instead of going over the objects folder
- Use mappings to find dmdSec for the files (more secure looking at fileId)
- Avoid saving IO three times
- Use 'p' instead of 's' for premis namespace"

jraddaoui committed:
Remove exiftoolRawOutput field from ES, refs #7823

sevein commented on plugins/arElasticSearchPlugin/lib/model/arElasticSearchInformationObjectPdo.class.php:
```diff
@@ -945,286 +945,31 @@ protected function getPhysicalObjects()
-          }
+    // Get METS filepath
+    $metsFilepath = sfConfig::get('sf_web_dir').
+      DIRECTORY_SEPARATOR.'uploads'.
```
Use sf_uploads_dir, because uploads/ doesn't have to be necessarily located under the root dir.

sevein commented:
Just one more thing. Do you mind to take a look at the headers when the client doesn't send cookies and it's hitting the API? For some reason back when we wrote this Symfony was sending Set-Cookie twice and I'm not sure why, but it's probably because one of the filters is maybe calling a function that we shouldn't be calling at that point.

jraddaoui committed:
Fix uploads paths, refs #7754

jraddaoui committed:
Fix Set-Cookie header in arRestApiPlugin responses

session_regenerate_id(true) deletes the old session file and submits a new session cookie
with the new session id, but it doesn't overwrite the old session values and the 'Set-Cookie'
header is sent twice (possible fix in PHP 5.6 -> https://github.com/php/php-src/pull/795)
In GET requests that perform login, the previous session needs to be destroyed before
calling session_regenerate_id(true) to fix this problem

Refs #7754

jraddaoui commented:
Merged!
